verdictd/app/src/PolicyEngine/opa/opaEngine.rs

```diff
@@ -0,0 +1,287 @@
+            Err(_) => println!("Failed to delete the wrong policy"),
+            Ok(_) => (),
+        };
+        return false;
+    }
+
+    true
+}
+
+/// Export existing policy from verdictd
+/// If the policy does not exist, an empty string will be returned
+pub fn export_policy(policy_name: &str) -> String {
+    let path = String::from(POLICY_PATH) + policy_name;
+    let mut contents = String::new();
+
+    // Open the file named policy_name
+    let mut file = match File::open(path) {
+        Err(_) => {
+            println!("Failed to open the policy");
+            return contents;
+        }
+        Ok(res) => res,
+    };
+
+    // Read the content of the policy to content
+    match file.read_to_string(&mut contents) {
+        Err(_) => {
+            println!("Failed to read the policy");
+            return contents;
+        }
+        Ok(res) => res,
+    };
+
+    contents
+}
+
+/// According to message and policy, the decision is made by opa
+/// message (JSON)
+/// {
+///     "mrEnclave" : "xxx"
+///     "mrSigner" : "xxx"
+///     "productId" : "xxx"
+///     ...
+/// }
+///
+/// returnValue(JSON)
+/// {
+///     "allow": true   
+///     "parserInfo": {
+///         "inputValue1": [
+///             "xxxxx",
+///             "xxxxx",
+///         ],
+///         "inputValue2": [
+///             "xxxxx",
+///             "xxxxx", ],
+///     }
+/// }
+pub fn make_decision(policy_name: &str, message: &str) -> String {
+    // Get the content of policy from policy_name
+    let policy = export_policy(policy_name);
+    let policy_go = GoString {
+        p: policy.as_ptr() as *const i8,
+        n: policy.len() as isize,
+    };
+
+    let message_go = GoString {
+        p: message.as_ptr() as *const i8,
+        n: message.len() as isize,
+    };
+
+    // Call the function exported by cgo and process the returned decision
+    let decision_buf: *mut c_char = unsafe { makeDecisionGo(policy_go, message_go) };
+    let decision_str: &CStr = unsafe { CStr::from_ptr(decision_buf) };
+    let decision_slice: &str = match decision_str.to_str() {
+        Err(_) => {
+            println!("Failed to get the decision");
+            return String::new();
+        }
+        Ok(res) => res,
+    };
+
+    decision_slice.to_string()
+}
+
+/// Write the string with the content of policy to the file named policy_name
+fn write_to_file(policy_name: &str, policy: &str) -> bool {
+    // Store the policy in the src/policy directory
+    let path = String::from(POLICY_PATH) + policy_name;
+    let path = Path::new(&path);
+
+    // Open the file in write-only mode
+    // If a policy with the same name already exists, it will be overwritten
+    let mut file = match File::create(&path) {
+        Err(_) => {
+            println!("Couldn't create file");
+            return false;
+        }
+        Ok(file) => file,
+    };
+
+    // Write the policy into file
+    match file.write_all(policy.as_bytes()) {
```

Comment:
Need to add a global read-write lock before write files.

Reply:
I have added a file lock, please review.